sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/UnwrapCastInBinaryComparisonSuite.scala

```diff
@@ -233,6 +233,15 @@ class UnwrapCastInBinaryComparisonSuite extends PlanTest with ExpressionEvalHelp
     assert(getRange(DecimalType(5, 2)).isEmpty)
   }
 
+  test("SPARK-35316: unwrap should support In/InSet predicate.") {
```

Comment:
can we have tests for the case when a `EqualTo` is converted to a `And` clause?

Reply:
Sure, The second one `Cast(f, LongType).in(1.toLong, Int.MaxValue.toLong)` is for this case.

Reply:
nit nit: perhaps remove `InSet` from `SPARK-35316: unwrap should support In/InSet predicate.` as we don't support it in this PR.